WordPress/gutenberg
# TextareaControl: Convert component to TypeScript

## What?
Converts the `TextareaControl` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
Refactors the current `TextareaControl` to TypeScript.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `TextareaControl` continues to function as expected
3. Test the various textarea controls within the block/site editors in Gutenberg still work
